src/classLibrary/ParsingStrategies/GuidResolver.ts

```diff
@@ -0,0 +1,96 @@
+import {
+    CyanSafe,
+    FileContent,
+    FileSystemInstance,
+    IGuidGenerator,
+    IParsingStrategy,
+    VirtualFileSystemInstance
+} from "../interfaces/interfaces";
+
+class GuidResolver implements IParsingStrategy {
+    private guidGenerator: IGuidGenerator;
+
+    constructor(guidGenerator: IGuidGenerator) {
+        this.guidGenerator = guidGenerator;
+    }
+
+    Count(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): Map<string, number> {
+        const guids: string[] = cyan.guid;
+        const result: Map<string, number> = new Map<string, number>();
+        virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            guids.Each((guid: string) => {
+                const count = this.CountKeyInVFS(guid, virtualFile);
+
+                if (result.has(guid)) {
+                    result.set(guid, result.get(guid)! + count);
+                } else {
+                    result.set(guid, count);
+                }
+            });
+        });
+
+        return result;
+    }
+
+    private CountKeyInVFS(key: string, virtualFile: VirtualFileSystemInstance): number {
+        return VirtualFileSystemInstance.match(virtualFile, {
+            File: (file: FileSystemInstance) => {
+                let tempCount = 0;
+                if (file["content"] != null && file.ignore.guidResolver.content) {
+                    FileContent.if.String(file.content, (str) => {
+                        tempCount += str.toUpperCase().Count(key.toUpperCase());
+                    });
+                }
+                return tempCount;
+            },
+            default: () => 0,
+        });
+    }
+
+    ResolveFiles(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+        return virtualFiles;
+    }
+
+    ResolveContents(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+        const guidsMap: Map<string, string> = cyan.guid.AsKey(() => this.guidGenerator.GenerateGuid());
+        return virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            VirtualFileSystemInstance.match(virtualFile, {
+                File: (file: FileSystemInstance) => {
+                    if (!file.ignore.guidResolver.content) return;
+                    if (file["content"] == null) return;
+                    FileContent.if.String(file.content, (str: string) => {
+                        guidsMap
+                            .Each((k: string, v: string) => {
+                                str = str.ReplaceAll(k.toLowerCase(), v).ReplaceAll(k.toUpperCase(), v);
+                            });
+                        file.content = FileContent.String(str);
+                    });
+                },
+                default: () => {
+                    return;
+                }
+            });
+        });
+    }
+
+    ReplaceGuid(guidArr: Map<string, string>, virtualFile: VirtualFileSystemInstance) {
```

Comment:
What's the difference between `ReplaceGuid` and `ResolveContents` O.o

Reply:
yeah honestly not much diff haha it does the same thing it just doesnt take in CyanSafe, saw it as a separate method in Ernest's repo so i included in case its used later but if not can remove

Reply:
Double check with you @kirinnee do we still need this method? It just looks like its replacing the placeholder GUID using some GuidArr

Reply:
wanted to extract out for testing so it doesn't depend on `CyanSafe`, ie lines 61 - 78 can be replaced with this method.

up to y'all, im ok with either